web/src/components/contributor-card.tsx

```diff
@@ -0,0 +1,82 @@
+import React from "react";
+import { Link } from "src/components/link";
+import { getContributorURL } from "src/utils/contributor";
+import { GetContributorsResponse } from "@dzcode.io/api/dist/contributor/types";
+
+export function ContributorCard({
+  contributor,
+  compact = false,
+  onClick,
+}: {
+  contributor: GetContributorsResponse["contributors"][number];
+  compact?: boolean;
+  onClick?: () => void;
+}) {
+  return (
+    <Link
+      className="card bg-base-300 w-full sm:max-w-52 rounded-lg flex flex-row border-base-200 border-2 overflow-hidden"
+      href={getContributorURL(contributor)}
+      onClick={onClick}
```

Comment:
Let's note this down:
```suggestion
      // TODO-OB: there's a bug here: when passing onClick to Link, the link no longer work as a SPA link, and instead causes a full reload of the page
      onClick={onClick}
```